langchain/tools/nuclia/tool.py

```diff
@@ -0,0 +1,185 @@
+"""Tool for the Nuclia Understanding API.
+
+Installation:
+
+```bash
+    pip install --upgrade protobuf
+    pip install nucliadb-protos
+```
+"""
+
+import base64
+import mimetypes
+import os
+from typing import Any, Dict, Optional, Type
+
+import requests
+from google.protobuf.json_format import MessageToJson
+from pydantic import BaseModel, Field
+
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForToolRun,
+    CallbackManagerForToolRun,
+)
+from langchain.tools.base import BaseTool
+
+try:
+    from nucliadb_protos.writer_pb2 import BrokerMessage
+except ImportError:
+    raise ImportError(
+        "nucliadb-protos is not installed. "
+        "Run `pip install nucliadb-protos` to install."
+    )
+
+
+class NUASchema(BaseModel):
+    action: str = Field(
+        ...,
+        description="Action to perform. Either `push` or `pull`.",
+    )
+    id: str = Field(
+        ...,
+        description="ID of the file to push or pull.",
+    )
+    enable_ml: bool = Field(
+        ...,
+        description="Enable Machine Learning processing (applicable only to `push` action).",
+    )
+    path: Optional[str] = Field(
+        ...,
+        description="Path to the file to push (needed only for `push` action).",
+    )
+
+
+class NucliaUnderstandingAPI(BaseTool):
+    """Tool to process files with the Nuclia Understanding API."""
+
+    name = "nuclia_understanding_api"
+    description = (
+        "A wrapper around Nuclia Understanding API endpoints. "
+        "Useful for when you need to extract text from any kind of files. "
+    )
+    args_schema: Type[BaseModel] = NUASchema
+    _results: Dict[str, Any] = {}
+
+    def _run(
+        self,
+        action: str,
+        id: str,
+        enable_ml: bool,
+        path: Optional[str],
+        run_manager: Optional[CallbackManagerForToolRun] = None,
+    ) -> str:
+        """Use the tool."""
+        zone = os.environ.get("NUCLIA_ZONE", "europe-1")
```

Comment:
could we pass zone and nua_key in during instantiation?

Reply:
right, fixed